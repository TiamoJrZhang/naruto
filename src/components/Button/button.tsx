/** @format */

import React, {FC} from 'react'
import {Button as AntButton} from 'antd'
import {ButtonProps} from 'antd/lib/button/index'

export interface BtnProps extends ButtonProps {
  status: number
}

const Button: FC<BtnProps> = props => {
  const {children, status} = props

  if (typeof status == 'string') {
    console.error('Warning: status can not be a string')
  }

  return (
    <div>
      <AntButton {...props}>{children && children}</AntButton>
    </div>
  )
}

export default Button
