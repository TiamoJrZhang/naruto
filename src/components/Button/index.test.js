/** @format */

import React from 'react'
import {mount} from 'enzyme'
import Button from './button'

describe('Button', () => {
  // mountTest(Button)
  // mountTest(() => <Button size='small'/>)
  // mountTest(() => <Button size='large'/>)

  it('renders correctly', () => {
    expect(mount(<Button>Follow</Button>)).toMatchSnapshot()
  })
  it('should warning when pass a string as status props', () => {
    //mockImplementation可以将console.error模拟为() => {}防止在控制台输出console.error
    const warnSpy = jest.spyOn(console, 'error').mockImplementation(() => void 0)
    mount(<Button size="small" status="ab" />)
    expect(warnSpy).toHaveBeenCalledWith('Warning: status can not be a string')
    warnSpy.mockRestore()
  })
  it('should support to change loading status', () => {
    const wrapper = mount(<Button>按钮</Button>)
    wrapper.setProps({loading: true})
    wrapper.update()
    expect(wrapper.find('.ant-btn-loading').length).toBe(1)
    wrapper.setProps({loading: false})
    wrapper.update()
    expect(wrapper.find('.ant-btn-loading').length).toBe(0)
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('should support link button', () => {
    const wrapper = mount(
      <Button target="_blank" href="http://ant.design">
        link button
      </Button>,
    )
    expect(wrapper.render()).toMatchSnapshot()
  })
})
