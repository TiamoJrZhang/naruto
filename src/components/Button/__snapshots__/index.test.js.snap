// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<Button>
  <div>
    <Button
      block={false}
      ghost={false}
      htmlType="button"
      loading={false}
    >
      <Wave>
        <button
          className="ant-btn"
          onClick={[Function]}
          type="button"
        >
          <LoadingIcon
            existIcon={false}
            loading={false}
            prefixCls="ant-btn"
          >
            <ForwardRef
              motionName="ant-btn-loading-icon-motion"
              onAppearActive={[Function]}
              onAppearStart={[Function]}
              onEnterActive={[Function]}
              onEnterStart={[Function]}
              onLeaveActive={[Function]}
              onLeaveStart={[Function]}
              removeOnLeave={true}
              visible={false}
            >
              <CSSMotion
                internalRef={null}
                motionAppear={true}
                motionEnter={true}
                motionLeave={true}
                motionName="ant-btn-loading-icon-motion"
                onAppearActive={[Function]}
                onAppearStart={[Function]}
                onEnterActive={[Function]}
                onEnterStart={[Function]}
                onLeaveActive={[Function]}
                onLeaveStart={[Function]}
                removeOnLeave={true}
                visible={false}
              />
            </ForwardRef>
          </LoadingIcon>
          <span
            key=".0"
          >
            Follow
          </span>
        </button>
      </Wave>
    </Button>
  </div>
</Button>
`;

exports[`Button should support link button 1`] = `
<div>
  <a
    class="ant-btn"
    href="http://ant.design"
    target="_blank"
  >
    <span>
      link button
    </span>
  </a>
</div>
`;
